taipy/core/_scheduler/_scheduler.py

```diff
@@ -174,18 +180,21 @@ def __unlock_edit_on_outputs(job):
     @classmethod
     def _on_status_change(cls, job: Job):
         if job.is_finished():
+            if job.is_cancelled():
+                cls.__unblock_jobs(if_canceled=True)
             cls.__unblock_jobs()
             cls.__execute_jobs()
 
     @classmethod
-    def __unblock_jobs(cls):
+    def __unblock_jobs(cls, if_canceled: bool = False):
         for job in cls.blocked_jobs:
             if not cls._is_blocked(job):
                 with cls.lock:
                     try:
-                        job.pending()
+                        if not if_canceled:
+                            job.pending()
+                            cls.jobs_to_run.put(job)
```

Comment:
I don't get why you need to change this in the __unblock_jobs method.

Reply:
because when you called job.cancelled(), it will trigger on_status_change which triggered __unblock_jobs, if you remove the cancelled job from blocked_jobs prior or after calling job.cancelled(), it will complain saying not being able to find the job